internal/elasticsearch/deployment.go

```diff
@@ -229,89 +224,61 @@ func (node *deploymentNode) unpause() error {
+	}
+	mutateFunc := func(current, _ *apps.Deployment) {
+		current.Spec.Paused = paused
+	}
 	// we use pauseNode so that we don't revert any new changes that should be made and
 	// noticed in state()
-	pauseNode := node.self.DeepCopy()
-	ll := log.WithValues("node", pauseNode.Name)
-
-	nretries := -1
-	retryErr := retry.RetryOnConflict(retry.DefaultRetry, func() error {
-		nretries++
-		if err := node.client.Get(context.TODO(), types.NamespacedName{Name: pauseNode.Name, Namespace: pauseNode.Namespace}, pauseNode); err != nil {
-			ll.Info("Could not get Elasticsearch node resource",
-				"error", err)
-			return err
-		}
-
-		if pauseNode.Spec.Paused == paused {
-			return nil
-		}
-
-		pauseNode.Spec.Paused = paused
-
-		if err := node.client.Update(context.TODO(), pauseNode); err != nil {
-			ll.Info("failed to update node resource",
-				"error", err)
-			return err
-		}
-		return nil
-	})
-	if retryErr != nil {
-		return kverrors.Wrap(retryErr, "could not update Elasticsearch node after retries",
-			"node", node.self.Name,
-			"retries", nretries,
+	pausedNode := node.self.DeepCopy()
+	err := deployment.Update(context.TODO(), node.client, pausedNode, equalFunc, mutateFunc)
+	if err != nil {
+		return kverrors.Wrap(err, "failed to update elasticsearch node deployment",
+			"cluster", node.clusterName,
+			"namespace", node.self.Namespace,
 		)
 	}
 
-	node.self.Spec.Paused = pauseNode.Spec.Paused
+	node.self.Spec.Paused = paused
 
 	return nil
 }
 
 func (node *deploymentNode) setReplicaCount(replicas int32) error {
-	nodeCopy := &apps.Deployment{}
-	nretries := -1
-	retryErr := retry.RetryOnConflict(retry.DefaultRetry, func() error {
-		nretries++
-		if err := node.client.Get(context.TODO(), types.NamespacedName{Name: node.self.Name, Namespace: node.self.Namespace}, nodeCopy); err != nil {
-			log.Info("Could not get Elasticsearch node resource, Retrying...", "error", err)
-			return err
-		}
-
-		if *nodeCopy.Spec.Replicas == replicas {
-			return nil
+	equalFunc := func(current, _ *apps.Deployment) bool {
+		if current.Spec.Replicas == nil {
+			return false
 		}
+		return *current.Spec.Replicas == replicas
+	}
+	mutateFunc := func(current, _ *apps.Deployment) {
+		current.Spec.Replicas = &replicas
+	}
 
-		nodeCopy.Spec.Replicas = &replicas
-
-		if err := node.client.Update(context.TODO(), nodeCopy); err != nil {
-			log.Info("failed to update node resource", "node", node.self.Name, "error", err)
-			return err
-		}
-
-		node.self.Spec.Replicas = &replicas
-
-		return nil
-	})
-	if retryErr != nil {
-		return kverrors.Wrap(retryErr, "could not update Elasticsearch node",
-			"node", node.self.Name,
-			"retries", nretries,
+	err := deployment.Update(context.TODO(), node.client, &node.self, equalFunc, mutateFunc)
+	if err != nil {
+		return kverrors.Wrap(err, "failed to update elasticsearch node deployment",
+			"cluster", node.clusterName,
+			"namespace", node.self.Namespace,
 		)
 	}
 
+	node.self.Spec.Replicas = &replicas
```

Comment:
do we need to set this if the update function does this for us?

Reply:
Same as above.